apps/fishing-map/features/map/annotations/Annotations.tsx

```diff
@@ -0,0 +1,85 @@
+import { useSelector } from 'react-redux'
+import cx from 'classnames'
+import { Popup } from 'react-map-gl'
+import { useTranslation } from 'react-i18next'
+import {
+  Button,
+  ColorBar,
+  IconButton,
+  InputText,
+  LineColorBarOptions,
+} from '@globalfishingwatch/ui-components'
+import {
+  useMapAnnotation,
+  useMapAnnotations,
+  DEFAUL_ANNOTATION_COLOR,
+} from 'features/map/annotations/annotations.hooks'
+import { useMapAnnotationDrag } from 'features/map/annotations/annotations-drag.hooks'
+import { selectMapAnnotation } from './annotations.slice'
+import styles from './Annotations.module.css'
+
+const colors = [{ id: 'white', value: DEFAUL_ANNOTATION_COLOR }, ...LineColorBarOptions]
+
+const MapAnnotations = () => {
+  useMapAnnotationDrag()
+  const { t } = useTranslation()
+  const mapAnnotation = useSelector(selectMapAnnotation)
+  const { resetMapAnnotation, setMapAnnotation } = useMapAnnotation()
+  const { deleteMapAnnotation, upsertMapAnnotations } = useMapAnnotations()
+
+  if (!mapAnnotation) {
+    return null
+  }
+
+  const onDeleteClick = () => {
+    deleteMapAnnotation(mapAnnotation.id)
+    resetMapAnnotation()
+  }
+
+  const onConfirmClick = () => {
+    upsertMapAnnotations({
+      ...mapAnnotation,
+      id: mapAnnotation.id || Date.now(),
+    })
+    resetMapAnnotation()
+  }
+
+  return (
+    <Popup
+      latitude={mapAnnotation.lat as number}
+      longitude={mapAnnotation.lon as number}
+      closeButton={true}
+      closeOnClick={false}
+      onClose={resetMapAnnotation}
+      maxWidth="330px"
+      className={cx(styles.popup)}
+    >
+      <div className={styles.popupContent}>
+        <div className={styles.flex}>
+          <InputText
+            value={mapAnnotation.label}
```

Comment:
Not sure if storing the `annotation.label` on the URL could eventually cause any issue due to too long URLs

Reply:
it could, but we don't have any other options if we want to make them shareable